src/proxy/cache/itemStore.js

```diff
@@ -28,32 +28,66 @@ import itemPreloaderFactory from 'taoQtiTest/runner/proxy/cache/itemPreloader';
 /**
  * The default number of items to store
  */
-var defaultConfig = {
+const defaultConfig = {
+    itemTTL: 0,
     maxSize: 10,
     preload: false
 };
 
 /**
  * Create an item store
- * @param {Object} [options]
- * @param {Number} [options.maxSize = 10] - the store limit
- * @param {Boolean} [options.preload] - do we preload items when storing them
- * @param {String} [options.testId] - the unique identifier of the test instance, required if preload is true
+ * @param {object} [options]
+ * @param {number} [options.itemTTL = 0] - The TTL for each item in the store, in milliseconds. 0 means no TTL.
+ * @param {number} [options.maxSize = 10] - the store limit
+ * @param {boolean} [options.preload] - do we preload items when storing them
+ * @param {string} [options.testId] - the unique identifier of the test instance, required if preload is true
  *
  * @returns {itemStore}
  */
 export default function itemStoreFactory(options) {
-    var config = _.defaults(options || {}, defaultConfig);
+    const config = _.defaults(options || {}, defaultConfig);
 
-    //in memory storage
-    var getStore = function getStore() {
-        return store('item-cache', store.backends.memory);
+    // in memory storage
+    const getStore = () => store('item-cache', store.backends.memory);
+
+    // maintain an index to resolve existence synchronously
+    const index = new Map();
+    let lastIndexedPosition = 0;
+
+    // check if a key has expired
+    const isExpired = key => {
+        const meta = index.get(key);
+        if (meta) {
+            return config.itemTTL && Date.now() - meta.timestamp >= config.itemTTL;
```

Comment:
In hypothetical case where asset implementation has no TTL and urls are permanent, what `config.itemTTL` do you expect to be used? `0` or very big number?